src/keyboard/mod.rs

```diff
@@ -124,14 +67,23 @@ impl Display for Key {
 }
 
 impl Key {
-    fn to_key_id(self) -> Result<u8> {
+    fn to_key_id_12(self) -> Result<u8> {
```

Comment:
This seems like a single method `to_key_id(max_keys: u8) -> Result<u8>`.
With comment explaining that `max_keys` is not actual number of keys on keyboard, but rather maximum number of keys this model is designed for.